strawberryfields/apps/data/feature.py

```diff
@@ -0,0 +1,231 @@
+    for each feature vector and the orbits/events used are also provided.
+
+    Attributes:
+        n_mean (float): mean number of photons used in the GBS device
+        threshold (bool): flag to indicate whether feature vectors are calculated with threshold
+            detectors or with photon-number-resolving detectors
+        method (str): method used to calculate the feature vectors; ``exact`` for exact calculation
+            or ``mc`` for Monte Carlo estimation
+        unit (str): signifies the unit of construction of feature vectors; ``orbits`` or ``events``
+        unit_data (list[list[int]]): list of orbits/events used to construct the feature vectors. Each
+            orbit is a list of integers and each event must be provided as
+            ``[total_photon_number, max_photon_per_mode]``
+        n_vectors (int): number of feature vectors provided in the dataset
+        n_features (int): number of features in each feature vector
+        vectors (array): array of feature vectors
+        adjs (array): array of adjacency matrices of graphs for which feature vectors
+            were calculated
+    """
+
+    _count = 0
+
+    @property
+    @abstractmethod
+    def _data_filename(self) -> str:
+        """Base name of files containing the data stored in the ``./feature_data/`` directory.
+
+        For each dataset, feature vectors and the corresponding adjacency matrices are provided
+        as numpy arrays in two separate ``.npy`` format files.
+
+        Given ``_data_filename = example`` and ``method = mc``, feature vectors should be stored
+        in ``./feature_data/example_mc_fv.npy`` and the array of corresponding adjacency matrices
+        should be saved in ``./feature_data/example_mat.npy``. Numpy functions such as
+        ``numpy.save(filename, array, allow_pickle=True)`` and
+        ``numpy.load(filename, allow_pickle=True)`` can be used to save and load these numpy
+        arrays."""
+        pass
+
+    # pylint: disable=missing-docstring
+    @property
+    @abstractmethod
+    def unit(self) -> str:
+        pass
+
+    # pylint: disable=missing-docstring
+    @property
+    @abstractmethod
+    def unit_data(self) -> list:
+        pass
+
+    # pylint: disable=missing-docstring
+    @property
+    @abstractmethod
+    def n_mean(self) -> float:
+        pass
+
+    # pylint: disable=missing-docstring
+    @property
+    @abstractmethod
+    def threshold(self) -> bool:
+        pass
+
+    # pylint: disable=missing-docstring
+    @property
+    @abstractmethod
+    def method(self) -> str:
+        pass
+
+    def __init__(self):
+        self.vectors = np.load(
+            f"{DATA_PATH}{self._data_filename}_{self.method}_fv.npy", allow_pickle=True
+        )
+        self.adjs = np.load(DATA_PATH + self._data_filename + "_mat.npy", allow_pickle=True)
+        self.n_vectors, self.n_features = self.vectors.shape
+
+    def __iter__(self):
+        return iter(self.vectors)

```

Comment:
```suggestion
        return self
```
This should work now after the updated tests

Reply:
Oh, I thought I already changed this. Thanks for checking again.